### DISCUSSION TITLE
Help with leds

### DISCUSSION THREAD

**Discussion by kilehh** [tttapa/Control-Surface#407] [Q&A] [answered]
Hello!

I have 6 74HC595 that im planing to use, and i want to use them for 22 leds and 10 leds which are meant for VU in total 32 leds. 
In examples i noticed that for mapping you need to "Map the Arduino as a Mackie Control Universal in your DAW." does that means it wont work in dj software, like virtualdj or mixxx etc? If it doesnt how i could fix that, or is it even possible to use leds with dj software?

Thanks in advance!

**Comment by tttapa** [comment, marked as answer]
Most audio applications I've used don't send any feedback back to the controller when you use the standard “MIDI learn mode”. On the other hand, when you select a commercial control surface in the settings, you do get feedback, but you're stuck with whatever protocol that  control surface uses.

If you can find a way to send feedback to the controller, e.g. the level meter values as control change messages, then you can display it to the LEDs using the Control Surface library. I haven't used Virtualdj and Mixxx in a long time, so I can't really tell you how to do the former, you'll have to dig into the documentation or the respective online fora.

**Reply by kilehh** [reply]
I read a bit today, there are is a way to make custom mapings using scripts in Virtualdj, i guess thats the only way to get it working.
Thanks!
